orchagent/txmonorch.cpp

```diff
@@ -0,0 +1,313 @@
+                    SWSS_LOG_INFO("TX_ERR threshold reset to %ld for port %s\n",
+                                  tesThreshold(tes), port.c_str());
+                }
+                else
+                {
+                    SWSS_LOG_ERROR("Unknown field type %s when handle threshold for %s\n",
+                                   fvField(i).c_str(), port.c_str());
+                    return -1;
+                }
+            }
+        }
+    }
+    catch (...) {
+        SWSS_LOG_ERROR("Fail to startTimer handle periodic update\n");
+    }
+
+    return 0;
+}
+
+/*handle configuration update*/
+void TxMonOrch::doTask(Consumer& consumer)
+{
+    int rc = 0;
+
+    SWSS_LOG_ENTER();
+    SWSS_LOG_INFO("TxMonOrch doTask consumer\n");
+
+    if (!gPortsOrch->allPortsReady())
+    {
+        SWSS_LOG_INFO("Ports not ready\n");
+        return;
+    }
+
+    auto it = consumer.m_toSync.begin();
+    while (it != consumer.m_toSync.end())
+    {
+        KeyOpFieldsValuesTuple t = it->second;
+
+        string key = kfvKey(t);
+        string op = kfvOp(t);
+        vector<FieldValueTuple> fvs = kfvFieldsValues(t);
+
+        rc = -1;
+
+        SWSS_LOG_INFO("TX_ERR %s operation %s set %s del %s\n",
+                      key.c_str(),
+                      op.c_str(), SET_COMMAND, DEL_COMMAND);
+        if (key == TXMONORCH_KEY_CFG_PERIOD)
+        {
+            if (op == SET_COMMAND)
+            {
+                rc = handlePeriodUpdate(fvs);
+            }
+            else
+            {
+                SWSS_LOG_ERROR("Unknown operation type %s when set period\n", op.c_str());
+            }
+        }
+        else //key should be the alias of interface
+        {
+            if (op == SET_COMMAND)
+            {
+                //fetch the value which reprsents threshold
+                rc = handleThresholdUpdate(key, fvs, false);
+            }
+            else if (op == DEL_COMMAND)
+            {
+                //reset to default
+                rc = handleThresholdUpdate(key, fvs, true);
+            }
+            else
+            {
+                SWSS_LOG_ERROR("Unknown operation type %s when set threshold\n", op.c_str());
+            }
+        }
+
+        if (rc)
+        {
+            SWSS_LOG_ERROR("Handle configuration update failed index %s\n", key.c_str());
+        }
+
+        consumer.m_toSync.erase(it++);
+    }
+}
+
+int TxMonOrch::pollOnePortErrorStatistics(const string &port, TxErrorStatistics &stat)
+{
+    uint64_t txErrStatistics = 0,
+        txErrStatLasttime = tesStatistics(stat),
+        txErrStatThreshold = tesThreshold(stat);
+    int tx_error_state,
+        tx_error_state_lasttime = tesState(stat);
+
+    SWSS_LOG_ENTER();
+
+    {
+        vector<FieldValueTuple> fvs;
+        std::stringstream stream;
+        stream << "oid:0x" << std::hex << tesPortId(stat);
+        std::string keystr( stream.str() );
+        m_countersTable.get(keystr, fvs);
```

Comment:
there is a simpler cleaner way of doing this:
 m_countersTable.get(sai_serialize_object_id(tesPortId(stat), fvs);

not needed all previous calls and vars

Reply:
I was looking for something like that.  I will try it.